pkg/fieldpath/paved.go

```diff
@@ -129,18 +131,21 @@ func getValueFromInterface(it any, s Segments) (any, error) {
 			}
 			it = array[current.Index]
 		case SegmentField:
-			object, ok := it.(map[string]any)
-			if !ok {
+			switch object := it.(type) {
+			case map[string]any:
+				v, ok := object[current.Field]
+				if !ok {
+					return nil, errNotFound{errors.Errorf("%s: no such field", s[:i+1])}
+				}
+				if final {
+					return v, nil
+				}
+				it = object[current.Field]
+			case nil:
+				return nil, errNotFound{errors.Errorf("path %q is not found in the paved object", s[:i])}
```

Comment:
Nit: Would it make sense to have it more consistent with the other error above? Something like `%s: expected map, got nil`

Reply:
Done!